libr/asm/arch/8051/8051_ass.c

```diff
@@ -413,22 +413,181 @@ static bool address_bit(char const* addr_str, ut8* addr_out) {
 	r_str_ncpy (bytepart, addr_str, separator - addr_str + 1);
 	bytepart[separator - addr_str + 1] = '\0';
 	r_str_ncpy (bitpart, separator + 1, strlen (separator));
-	if (!address_direct (bytepart, &byte)) {
-		goto end;
-	}
-	if (1 < strlen (bitpart)
-		|| bitpart[0] < '0' || '7' < bitpart[0]) {
-		ret = false;
-		goto end;
-	}
-	bit = bitpart[0] - '0';
-	if (0x20 <= byte && byte < 0x30) {
-		*addr_out = (byte - 0x20) * 8 + bit;
-		ret = true;
-	} else if (0x80 <= byte && !(byte%8)) {
-		*addr_out = byte + bit;
-		ret = true;
+
+	if (!strcmp (bytepart, "b")) {
+		if ((bitpart[0] - '0') <= 9) {
+			*addr_out = 0xf0 + (bitpart[0] - '0');
```

Comment:
as the structure is the same in each place here, it may be easier to read if you put the constant part (`0xf0`) into a variable, then have one place where you set `*addr_out` after the loop

Reply:
@Lazula  could you provide an example please?

Reply:
something like this:
```
ut8 addr_bytepart;
if (!strcmp (bytepart, "b")) {
        addr_bytepart = 0xf0;
}
```

and at the end:

```
*addr_out = addr_bytepart + bit;
```

Reply:
fine I understand